_meeting-notes/2019-10-22.md

```diff
@@ -0,0 +1,128 @@
+
+## :question: Questions (Blockers, Technical Design needs)
+
+- (@jimpick) - Error handling patterns - which ones?
+  - panic(err)
+  - runenv.Abort(err)
+  - log.Fatal("error message", err)
+  - return result, err
+  - just exit process?
+  - expected number of tests? eg. TAP (test anything protocol) http://testanything.org/tap-specification.html
+  - Example code: https://github.com/ipfs/testground/blob/plan-smlbench-two-containers/plans/smlbench2/main.go
+  - A:
+    - Option:
+      - A) Given that we are paying to run these tests, once a test fails, we should just shutdown the whole thing
+      - B) Differentiate types of error (e.g. fatal & error), one goes through path A, the other gets gathered by the same pipeline that grabs metrics and takes them to Elastic Search
+      - C) Rely on another process (the main orchestrator) to make the decision to stop or continue depending on type of test
+        - Dependency: Needs the TestGround daemon done first 
+- (@jimpick) - How to organize tests that run multiple times with different file sizes?
+  - A: It lives within the Test Parameters
+  - Follow up Question:
+    - Where does the decision of running the test multiple times happens? Options:
+      - A) In the Orchestrator (loop outside)
+        - Steven prefers this one. Ideally there should be a matrix of parameters.
+      - B) In the Test Case itself (loop inside)
+      - 
+- (@jimpick) - Tradeoff questions:
+  - Tests that run scenarios (even if ugly) vs. clean looking tests?
+  - How much code review?
+    - Always for code. 
+  - Refactoring into SDK?
+    - For Q4, good to go ugly, 2020 Q1, reconsider and/or refactor
+- (@daviddias) - On onboarding more folks, we should we go after? :D
+  - Dirk? -> Bitswap related Test Plans
+  - Adin? -> DHT related Test Plans
+  - Alex? -> Not for now
+  - ?? Yes to write the Test Plan themselves, not the TestGround code
+
+## :dart: Next
+
+### OKRs for the Quarter
+
+![](https://raw.githubusercontent.com/ipfs/testground/master/docs/img/Q4OKRs.png)
+
+### Top 3 things to get done this week
+
+- Finalize the spec for the 10 Test Plans
+- Land [Plan: Data Exchange with Datasets of Interest (BitSwap/GraphSync)](https://github.com/ipfs/testground/blob/master/docs/tests-mvp.md#plan-data-exchange-with-datasets-of-interest-bitswapgraphsync) end-to-end (Plan, Tests & Graphs) so that we can onboard Dirk that needs it for testing Bitswap changes
+- ~~Rope in Adin by onboarding him to the Network Connectivity Test Plans (needs the spec to be finalized and an example using TC done)~~ Given the limited amount of time, we might not be able to do it this week and only on the next.
+
+### Assignments
+
+- ==@stebalien==
+  - `P0` Reach out to Dirk to see if he can review the Test Plan https://github.com/ipfs/testground/tree/master/plans/data-transfer-datasets-random
+  - Review all Test Plan https://github.com/ipfs/testground/blob/master/docs/PLAN_SPEC.md
```

Comment:
Same as above  https://trello.com/c/XINMtY3S/29-what-are-the-exact-change-sets-0422v050-that-wed-like-to-test